### PR TITLE
Moved (or rather copied) the option "Enable HTTP state" to options

### PR DESCRIPTION
**Pull request by rain0r**
See https://github.com/zaproxy/zaproxy/issues/3394

### CODE REVIEW

**Comment by rain0r** [comment]
Still to do: Removing the option from the "Edit" menu

**Comment by thc202** [comment]
Do we want to remove it from the menu?

File: src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java [hunk 1]
```diff
@@ -818,16 +821,17 @@ private JPanel getPanelGeneral() {
 			gbc.fill = java.awt.GridBagConstraints.HORIZONTAL;
 			gbc.insets = new java.awt.Insets(2,2,2,2);
 			gbc.anchor = java.awt.GridBagConstraints.WEST;
-            
+
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java]: Formatting changes should be done in a separate PR.
https://github.com/zaproxy/zaproxy/wiki/DevGuidelines#formatting

File: src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java [hunk 2]
```diff
@@ -818,16 +821,17 @@ private JPanel getPanelGeneral() {
 			gbc.fill = java.awt.GridBagConstraints.HORIZONTAL;
 			gbc.insets = new java.awt.Insets(2,2,2,2);
 			gbc.anchor = java.awt.GridBagConstraints.WEST;
-            
+
 			JLabel uaLabel = new JLabel(Constant.messages.getString("conn.options.defaultUserAgent"));
 			uaLabel.setLabelFor(this.getDefaultUserAgent());
 			panelGeneral.add(uaLabel, LayoutHelper.getGBC(0, 1, 1,0.5D));
-			panelGeneral.add(this.getCommonUserAgents(), 
+			panelGeneral.add(getCommonUserAgents(),
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java]: Was this change actually necessary as part of this PR? (Same just below)

File: src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java [hunk 3]
```diff
@@ -854,7 +858,18 @@ private JCheckBox getCheckBoxSingleCookeRequestHeader() {
         return checkBoxSingleCookieRequestHeader;
     }
 
-    private ProxyExcludedDomainsMultipleOptionsPanel getProxyExcludedDomainsPanel() {
+    public JCheckBox getCheckBoxHttpStateEnabled() {
+    	if (checkBoxHttpStateEnabled == null) {
+    		checkBoxHttpStateEnabled = new JCheckBox(Constant.messages.getString("conn.options.httpStateEnabled"));
+    	}
+		return checkBoxHttpStateEnabled;
+	}
```
Comment (kingthorin) [inline src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java]: Seems to be something weird with the indentation here, I'm guessing there's mixed spaces and tabs.

File: src/org/parosproxy/paros/network/ConnectionParam.java [hunk 4]
```diff
@@ -223,6 +225,13 @@ protected void parse() {
         }
         
         try {
+            this.httpStateEnabled = getConfig().getBoolean(HTTP_STATE_ENABLED, true);
+        } catch (ConversionException e) {
+            log.error("Error while loading the option httpStateEnabled: " + e.getMessage(), e);
+        }
+        
+
```
Comment (kingthorin) [inline src/org/parosproxy/paros/network/ConnectionParam.java]: Seems to be something weird with the indentation here, I'm guessing there's mixed spaces and tabs.

Also the extra blank line should be removed.

File: src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java (hunk 3 above)
Comment (rain0r) [inline src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java]: Can't wait for https://github.com/zaproxy/zap-api-java/pull/32 to come in place ...

File: src/org/parosproxy/paros/network/ConnectionParam.java (hunk 4 above)
Comment (rain0r) [inline src/org/parosproxy/paros/network/ConnectionParam.java]: Can't wait for https://github.com/zaproxy/zap-api-java/pull/32 to come in place ...

File: src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java (hunk 2 above)
Comment (rain0r) [inline src/org/parosproxy/paros/extension/option/OptionsConnectionPanel.java]: No. Just wanted to align it to the rest of the source code.

File: src/org/parosproxy/paros/network/ConnectionParam.java [hunk 5]
```diff
@@ -223,6 +225,13 @@ protected void parse() {
         }
         
         try {
+            this.httpStateEnabled = getConfig().getBoolean(HTTP_STATE_ENABLED, true);
```
Comment (thc202) [inline src/org/parosproxy/paros/network/ConnectionParam.java]: The default should be `false` (currently that's not enabled by default).

Comment (rain0r) [inline src/org/parosproxy/paros/network/ConnectionParam.java]: Changed it. See https://github.com/zaproxy/zaproxy/pull/3422/commits/f823b8f4af36c52cb8080a6157afea4611d3a9f2

**Comment by kingthorin** [comment]
You seem to be mixing branches....

**Comment by rain0r** [comment]
Is it possible for you to join `#websectools`. Communication might be easier there.
